main/src/main/scala/sbt/Defaults.scala

```diff
@@ -202,7 +203,40 @@ object Defaults extends BuildCommon {
           ++ Vector(ServerHandler.fallback))
       },
       uncachedStamper := Stamps.uncachedStamps(fileConverter.value),
-      reusableStamper := Stamps.timeWrapLibraryStamps(uncachedStamper.value, fileConverter.value),
+      reusableStamper := {
+        val converter = fileConverter.value
+        val unmanagedCache = unmanagedFileStampCache.value
+        val managedCache = managedFileStampCache.value
+        val backing =
+          Stamps.timeWrapLibraryStamps(uncachedStamper.value, fileConverter.value)
+        new xsbti.compile.analysis.ReadStamps {
```

Comment:
extract to a named dedicated class?

Reply:
What is the use case for using a dedicated class outside of this method? It uses `private[sbt]` classes so I can't see such a class being public and I'm not sure where else in sbt it would be used.